docs/site/Parsing-requests.md

```diff
@@ -234,6 +234,76 @@ class Product extends Entity {
 A full list of options & usage scenarios could be found in the
 [documentation of AJV errors](https://github.com/epoberezkin/ajv-errors).
 
+##### Custom AJV Formats
+
+AJV Custom Formats can be configured by binding the value
+`validation.ajvCustomFormats` to `RestBindings.REQUEST_BODY_PARSER_OPTIONS`
+('rest.requestBodyParserOptions').
+
+For example,
+
+```ts
+server.bind(RestBindings.REQUEST_BODY_PARSER_OPTIONS).to({
+  validation: {
+    ajvCustomFormats: {
+      'only-alphanumeric': /^[a-zA-Z0-9 ]+$/.source,
+      'product-code': {
+        async: true,
```

Comment:
I think asynchronous detection should be automated (Possible solution: https://stackoverflow.com/questions/38508420/how-to-know-if-a-function-is-async). Having a manual flag is room for human error.

Reply:
An input to custom format could be string, regexp, function or an object altogether as explained at https://ajv.js.org/#api-addformat.

Although the async function can be detected, I believe doing it all will add unnecessary logic on framework side. ajv anyway does all these validations.

Plus, at times, too much automation might not be a good choice.

Reply:
IMO it there isn't much code added to detect async, and it's a better tradeoff than unexpected behaviour due to a separate flag that was set incorrectly.

Reply:
Not very convinced by `async: true`. It feels redundant when we are already specifying the validation function is async.